xtuner/v1/train/trainer.py

```diff
@@ -11,6 +11,16 @@
 from shutil import rmtree
 from typing import Annotated, Literal, Sized, cast
 
+
+try:
```

Comment:
Suggest using `device=="cuda"` to determine whether bind cpu by numa, and Add `numa` and `pynvml_utils` to pyproject.toml